sycl/test/kernel_param/attr.cpp

```diff
@@ -0,0 +1,22 @@
+// RUN: %clangxx -fsycl-device-only -Xclang -fsycl-is-device -emit-llvm %s -S -o %t.ll -I %sycl_include -Wno-sycl-strict -Xclang -verify-ignore-unexpected=note,warning -Xclang -disable-llvm-passes -D__ENABLE_USM_ADDR_SPACE__
```

Comment:
Your changes are changing the frontend, so I'd expect to see these tests in clang\test\CodeGenSYCL (since it's testing codegen). However, I'm not super familiar with how we're testing SYCL functionality so please push back if you think I'm wrong here.

Reply:
The tests in clang/test/codeGenSYCL use "fake" SYCL headers. The support we are adding here is for a real transformation made in the SYCL handler.hpp header, so it needs to be tested with the real handler.hpp.

Reply:
Please correct me if I am mistaken, but can't a FE test be done with the wrapper defined manually in test (i.e. write a FE test with an already transformed kernel). If I understand this correctly, any kernel called "__pf_kernel_wrapper" should trigger this code.